MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionViewQtGUI.cpp

```diff
@@ -1261,6 +1300,163 @@ void EnggDiffractionViewQtGUI::setListWidgetBank(int idx) {
   selectBank->setCurrentRow(idx);
 }
 
+void MantidQt::CustomInterfaces::EnggDiffractionViewQtGUI::
+    fittingRunNoChanged() {
+  try {
+    QString focusedFile = m_uiTabFitting.lineEdit_pushButton_run_num->text();
+    std::string strFocusedFile = focusedFile.toStdString();
+    // file name
+    Poco::Path selectedfPath(strFocusedFile);
+    Poco::Path bankDir;
+
+    // handling of vectors
+    m_fitting_runno_dir_vec.clear();
+    std::string strFPath = selectedfPath.toString();
+    std::vector<std::string> splitBaseName = splitFittingDirectory(strFPath);
+
+    if (selectedfPath.isFile() && !splitBaseName.empty()) {
+
+#ifdef __unix__
+      auto home = Poco::Path().home();
+      bankDir.append(home);
+      bankDir.append(selectedfPath.parent().toString());
+      bankDir.append(selectedfPath);
+#else
+      bankDir = (bankDir).expand(selectedfPath.parent().toString());
+#endif
+
+      if (!splitBaseName.empty() && splitBaseName.size() > 3) {
+        std::string foc_file = splitBaseName[0] + "_" + splitBaseName[1] + "_" +
+                               splitBaseName[2] + "_" + splitBaseName[3];
+        std::string strBankDir = bankDir.toString();
+        updateFittingDirVec(strBankDir, foc_file);
+      }
+      // if run number length greater
+    } else if (focusedFile.count() > 4) {
+      // if given a run number instead
+      updateFittingDirVec(m_focusDir, strFocusedFile);
+    } else {
+      userWarning("Invalid Input", "Invalid directory or run number given. "
+                                   "Please try again");
+    }
+
+    try {
+      // add bank to the combo-box and list view
+      addBankItems(splitBaseName, focusedFile);
+    } catch (...) {
+      userWarning("Unable to insert items: ", "Could not add banks to "
+                                              "combo-box or list widget. "
+                                              "Please try again");
+    }
+  } catch (...) {
+    userWarning("Invalid file", "Unable to select the file, please check the "
+                                "directory or the file name and try again");
```

Comment:
This type of error message can be very arcane when an unexpected error happens. I'd add more information (especially the string with the path that you were trying to look for). Then the source of the error becomes obvious to the user (hopefully).